datafusion/core/src/logical_plan/builder.rs

```diff
@@ -191,16 +191,16 @@ impl LogicalPlanBuilder {
     /// Convert a table provider into a builder with a TableScan
     pub fn scan(
         table_name: impl Into<String>,
-        provider: Arc<dyn TableProvider>,
+        table_source: Arc<dyn TableSource>,
         projection: Option<Vec<usize>>,
     ) -> Result<Self> {
-        Self::scan_with_filters(table_name, provider, projection, vec![])
+        Self::scan_with_filters(table_name, table_source, projection, vec![])
     }
 
     /// Convert a table provider into a builder with a TableScan
     pub fn scan_with_filters(
         table_name: impl Into<String>,
-        provider: Arc<dyn TableProvider>,
+        table_source: Arc<dyn TableSource>,
```

Comment:
I wonder if we could make the APIs a little  (avoid having to call `provider_as_source`) nicer with something like

Untested:
```
table_source: impl Into<Arc<dyn TableSource>>
```

Or add some other trait that would allow us to pass both `Arc<dyn TableSource>` as well as `Arc<dyn TableProvider>`

I think we can always refine the API in a follow on PR